lib/internal/http2/compat.js

```diff
@@ -850,29 +849,14 @@ class Http2ServerResponse extends Stream {
   writeEarlyHints(hints) {
     validateObject(hints, 'hints');
 
-    const headers = { __proto__: null };
-
-    const linkHeaderValue = validateLinkHeaderValue(hints.link);
-
-    for (const key of ObjectKeys(hints)) {
-      if (key !== 'link') {
-        headers[key] = hints[key];
-      }
-    }
-
-    if (linkHeaderValue.length === 0) {
-      return false;
-    }
-
     const stream = this[kStream];
 
     if (stream.headersSent || this[kState].closed)
       return false;
 
     stream.additionalHeaders({
-      ...headers,
+      ...hints,
```

Comment:
This **might** have unintended consequences

Reply:
I don't think so, the spread operator does not traverse the prototype chain:

```js
> a = { foo: 1 }
{ foo: 1 }
> b = Object.create(a)
{}
> Object.keys(b)
[]
> { ...b }
{}
>
```